src/unit.rs

```diff
@@ -127,8 +130,9 @@ impl<'input, Endian> CompilationUnitHeadersIter<'input, Endian>
         if self.input.is_empty() {
             Ok(None)
         } else {
-            match CompilationUnitHeader::parse(self.input) {
+            match CompilationUnitHeader::parse(self.input, self.offset) {
                 Ok((rest, header)) => {
+                    self.offset.0 += self.input.len() as u64 - rest.len() as u64;
```

Comment:
Did you happen to bench before and after adding offsets into units? I expect it should fall within the noise, and I think we want this change either way, but I am a little curious.